pkg/querier/astmapper/parallel.go

```diff
@@ -42,7 +42,17 @@ func CanParallelize(node promql.Node) bool {
 
 	case *promql.AggregateExpr:
 		_, ok := summableAggregates[n.Op]
-		return ok && CanParallelize(n.Expr)
+		if !ok {
+			return ok
```

Comment:
Just for the sake of clarity
```suggestion
			return false
```